assets/scripts/palette/Palette.jsx

```diff
@@ -28,63 +23,54 @@ function Palette (props) {
     }
   }, [])
 
-  function renderPaletteItems () {
-    const segments = getAllSegmentInfoArray()
-
-    // For each segment, set "disabled" property instead that indicates
-    // whether this segment is in a disabled state for this user
-    // Then filter out disabled segments that do not have the
-    // `alwaysShowInPalette` property set to `true`
-    const displayedSegments = segments
-      .map((segment) => {
-        // Accept segments that don't have the `enableWithFlag` property
-        const enabledByDefault = !segment.enableWithFlag
-        // Accept segments with the `enableWithFlag` property, but only if
-        // the flags have that value set to true.
-        const enabledByFlag =
-          segment.enableWithFlag && flags[segment.enableWithFlag].value
+  const segments = getAllSegmentInfoArray()
 
-        return {
-          ...segment,
-          disabled: !(enabledByDefault || enabledByFlag)
-        }
-      })
-      .filter(
-        (segment) =>
-          !segment.disabled || (segment.disabled && segment.alwaysShowInPalette)
-      )
-
-    // Return all enabled segments as an array of SegmentForPalette components.
-    return displayedSegments.map((segment) => {
-      const variant = segment.paletteIcon
-        ? segment.paletteIcon
-        : Object.keys(segment.details).shift()
+  // For each segment, set "disabled" property instead that indicates
```

Comment:
hah, this is a doozy, thanks for the code comment! Would this count as a legacy issue that might be simplified in the future?

Reply:
This is actually new-ish behavior, but it did grow out of various new additions over the past few years. Open to recommendations for change.